include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -286,6 +294,63 @@ __pattern_adjacent_find(_ExecutionPolicy&& __exec, _Range&& __rng, _BinaryPredic
     return return_value(result, __rng.size(), __is__or_semantic);
 }
 
+//------------------------------------------------------------------------
+// merge
+//------------------------------------------------------------------------
+
+template <typename _ExecutionPolicy, typename _Range1, typename _Range2, typename _Range3, typename _Compare>
+oneapi::dpl::__internal::__enable_if_hetero_execution_policy<_ExecutionPolicy,
+                                                             oneapi::dpl::__internal::__difference_t<_Range3>>
+__pattern_merge(_ExecutionPolicy&& __exec, _Range1&& __rng1, _Range2&& __rng2, _Range3&& __rng3, _Compare __comp)
+{
+    auto __n1 = __rng1.size();
+    auto __n2 = __rng2.size();
+    auto __n = __n1 + __n2;
+    if (__n == 0)
```

Comment:
Please note that the following test case:
```c++
#include "support/pstl_test_config.h"

#include _PSTL_TEST_HEADER(execution)
#if _ONEDPL_USE_RANGES
#include _PSTL_TEST_HEADER(ranges)
#endif

#include "support/utils.h"
#include <iostream>

int32_t
main()
{
#if _ONEDPL_USE_RANGES
    using T = int;

    std::vector<T> in;

    using namespace TestUtils;
    using namespace oneapi::dpl::experimental::ranges;
    {
        sycl::buffer<T> C(in.data(), sycl::range<1>(in.size()));
        std::cout << "all_view(C).size(): " << all_view(C).size() << std::endl;
    }
#endif //_ONEDPL_USE_RANGES
    ::std::cout << TestUtils::done() << ::std::endl;
    return 0;
}
```
fails with:
```c++
terminate called after throwing an instance of 'cl::sycl::invalid_object_error'
  what():  SYCL buffer size is zero. To create a device accessor, SYCL buffer size must be greater than zero. -30 (CL_INVALID_VALUE)
```

That means that the following checks will never trigger because we are not able to create a zero-sized range from a zero-sized buffer.

Reply:
This may be a more general issue, thus not a subject of this patch.

Reply:
About "if (__n == 0)" and "never trigged"  - essentially, we may have a range which is based on a range factory(iota, etc), non-sycl_view over sycl buffer.